pkg/buses/gcppubsub/bus.go

```diff
@@ -284,9 +285,18 @@ func (b *CloudPubSubBus) deleteSubscription(ref buses.SubscriptionReference) err
 }
 
 func (b *CloudPubSubBus) topicID(channel buses.ChannelReference) string {
-	return fmt.Sprintf("channel-%s-%s-%s", b.ref.Name, channel.Namespace, channel.Name)
+	return fmt.Sprintf("channel-%s-%s-%s", channel.Namespace, channel.Name, b.id)
 }
 
 func (b *CloudPubSubBus) subscriptionID(subscription buses.SubscriptionReference) string {
-	return fmt.Sprintf("subscription-%s-%s-%s", b.ref.Name, subscription.Namespace, subscription.Name)
+	return fmt.Sprintf("subscription-%s-%s-%s", subscription.Namespace, subscription.Name, b.id)
+}
+
+// busId generates a short unique identifier for a bus from the Bus's UID.
+// This value is used as a component of the topic and subscription IDs
+func busId(uid string) string {
```

Comment:
There is no test added for this method. Please add a test to make sure this does not make the same busID for similar UIDs